core/service/src/lib.rs

```diff
@@ -81,7 +81,7 @@ pub struct Service<Components: components::Components> {
 		NetworkStatus<ComponentBlock<Components>>, NetworkState
 	)>>>>,
 	transaction_pool: Arc<TransactionPool<Components::TransactionPoolApi>>,
-	keystore: Option<Arc<Mutex<Store>>>,
+	pub keystore: Arc<Mutex<Store>>,
```

Comment:
Why `pub`?

Reply:
To avoid adding all store methods to service. They are all used (exclusively) in the AuthoritySetup part of node/cli/src/service.rs

Reply:
This is extremely bad, but not easy to do properly without refactoring `components.rs`
Please at least add a `lock_keystore()` method on the service that returns a lock to the keystore, instead of having public fields.